### PR TITLE
Fix HomeFeedView Previews

### PR DESCRIPTION
**Pull request by mplorentz**
## Issues covered
This fixes the SwiftUI preview so I can work on #1291

## Description
The preview was crashing in `KeyChain.swift`. I'm not sure exactly why but we shouldn't be trying to write to the real keychain in previews anyway, so I moved the keychain code into our dependency injection system and made a `MockKeychain` that just stores values in memory that we can use for previews.

Then I had to do a little refactoring of the way `PreviewData` instantiates the `CurrentUser` to get the preview working.

## How to test
Make sure the live app still loads keys correctly, and verify that the HomeFeedView and other modified previews work.

### CODE REVIEW

File: Nos.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -149,10 +149,10 @@
 		A336DD3C299FD78000A0CBA0 /* Filter.swift in Sources */ = {isa = PBXBuildFile; fileRef = A336DD3B299FD78000A0CBA0 /* Filter.swift */; };
 		A34E439929A522F20057AFCB /* CurrentUser.swift in Sources */ = {isa = PBXBuildFile; fileRef = A34E439829A522F20057AFCB /* CurrentUser.swift */; };
 		A351E1A229BA92240009B7F6 /* ProfileEditView.swift in Sources */ = {isa = PBXBuildFile; fileRef = A351E1A129BA92240009B7F6 /* ProfileEditView.swift */; };
-		A3B943CF299AE00100A15A08 /* KeyChain.swift in Sources */ = {isa = PBXBuildFile; fileRef = A3B943CE299AE00100A15A08 /* KeyChain.swift */; };
+		A3B943CF299AE00100A15A08 /* Keychain.swift in Sources */ = {isa = PBXBuildFile; fileRef = A3B943CE299AE00100A15A08 /* Keychain.swift */; };
```
Comment (joshuatbrown) [inline Nos.xcodeproj/project.pbxproj:152]: 😍

File: Nos/Service/Keychain.swift [hunk 2]
```diff
@@ -50,3 +50,22 @@ enum KeyChain {
         return SecItemDelete(query as CFDictionary)
     }
 }
+
+class MockKeychain: Keychain {
```
Comment (joshuatbrown) [inline Nos/Service/Keychain.swift:54]: I think I just realized why I prefer not subclassing for mocks: if we add a new func to Keychain that does something real but forget to override it in the Mock to do something fake, now we have a MockKeychain that does something real. Would a protocol or [the pointfree approach](https://pointfreeco.github.io/swift-dependencies/main/documentation/dependencies/designingdependencies) work, or do those approaches just make everything too messy?

File: Nos/Views/Home/HomeFeedView.swift [hunk 3]
```diff
@@ -188,57 +188,14 @@ struct HomeFeedView: View {
     }
 }
 
-struct ContentView_Previews: PreviewProvider {
+#Preview {
```
Comment (joshuatbrown) [inline Nos/Views/Home/HomeFeedView.swift:191]: 💯 glad we're on the same page about replacing `PreviewProvider` with `#Preview`.

**Review by joshuatbrown — COMMENTED** [review]
I fully support this direction.

File: Nos/Service/Keychain.swift (hunk 2 above)
Comment (mplorentz) [reply Nos/Service/Keychain.swift:54]: Yeah, a protocol is just better. I think I'm still conditioned by the days when generics would just not be usable in random places, but now that we have existential types I don't think that's the case. I will change it.

**Pull request merged**